caas/kubernetes/provider/cloud.go

```diff
@@ -227,10 +229,17 @@ func BaseKubeCloudOpenParams(cloud jujucloud.Cloud, credential jujucloud.Credent
 
 // FinalizeCloud is part of the environs.CloudFinalizer interface.
 func (p kubernetesEnvironProvider) FinalizeCloud(ctx environs.FinalizeCloudContext, cld cloud.Cloud) (cloud.Cloud, error) {
-	cloudName := cld.Name
-	if cloudName != caas.K8sCloudMicrok8s {
+	if cld.Name != caas.K8sCloudMicrok8s {
 		return cld, nil
 	}
+
+	if cld.Config["operator-storage"] != nil && cld.Config["workload-storage"] != nil {
+		return cld, nil
+	}
+
+	if !microK8sStorageEnabled(p.cmdRunner) {
```

Comment:
They need "dns" enabled too, would be awesome if we could check for that as well.
Also, these dns and storage enabled checks need to be before the cld.Config check since even if the cloud is configured etc, it will fail without storage enabled.